idea-plugin/src/main/kotlin/arrow/meta/ide/phases/resolve/QuoteSystem.kt

```diff
@@ -0,0 +1,267 @@
+        /*if (transformed.isNotEmpty()) {
+          if (!ApplicationManager.getApplication().isUnitTestMode) {
+            ApplicationManager.getApplication().invokeLater {
+              WriteAction.run<Exception> {
+                transformed.forEach { (original, transformed) ->
+                  val metaFile = original.virtualFile.parent.findOrCreateChildData(this, transformed.name + ".txt")
+                  metaFile.setBinaryContent(transformed.text.toByteArray())
+                }
+              }
+            }
+          }
+        }*/
+
+        // update the resolved data as soon as index access is possible
+        // fixme protect against multiple transformations at once
+        // fixme this might be delayed, make sure we're handling this correctly with a test
+        DumbService.getInstance(project).runReadActionInSmartMode {
+          if (strategy.resetCache) {
+            cache.clearCache()
+          }
+          LOG.info("resolving descriptors of transformed files: ${cache.cache.size} files")
+          // clear descriptors of all updatedFiles, which don't have a transformation result
+          // e.g. because no meta-code is used anymore
+          // fixme atm a transformation of a .kt file with syntax errors also returns an empty list of transformations
+          //    we probably need to handle this, otherwise files with errors will always have unresolved references
+          //    best would be partial transformation results for the valid parts of a file (Quote system changes needed)
+
+          // fixme this lookup is slow (exponential?), optimize when necesary
+          val removedQuotes: List<QuotedFile> =
+            files
+              .filter { src -> !transformed.any { it.first == src } }
+              .mapNotNull { src -> cache.removeQuotedFile(src) }
+          val facade: ResolutionFacade = KotlinCacheService.getInstance(project).getResolutionFacade(files + transformed.map { it.second })
+          val resolvedFiles: List<Pair<KtFile, QuotedFile>> =
+            transformed
+              .map { (origin, quoted) ->
+                origin to
+                  quoted.resolve(facade, BodyResolveMode.FULL).run { copy(second = second.filter { it.isMetaSynthetic() }) }
+              }
+
+          // fixme this triggers a call to the .resolved()
+          // via MetaSyntheticPackageFragmentProvider.BuildCachePackageFragmentDescriptor.Scope.getContributedClassifier
+          resolvedFiles
+            .map { (_, quoted) -> quoted.second }
+            .flatten()
+            .forEach { descriptor ->
+              try {
+                descriptor.safeAs<LazyEntity>()?.forceResolveAllContents()
+              } catch (e: IndexNotReadyException) {
+                LOG.warn("Index wasn't ready to resolve: ${descriptor.name}")
+              }
+            }
+
+          transformed.forEach { // refresh the highlighting of editors of modified files, using the new cache
```

Comment:
A new file which also modifies some of the extracted code is a bit hard to review. 
The old code had `metaCache.updateTransformations(sourceFile, transformedFile, synthDescriptors)` after the calls of `forceResolveAllContents()`. I don't see it anymore or am I missing something?

Reply:
Yes the `update` is added in the next commit